api/proto/hapi/service-event.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/chanzuckerberg/happy/api/events";
+
+package happy.api.events;
+
+message Event {
+
+    string event_type = 2; // Free-form, but should be consistent and SCREAMING_SNAKE_CASE
+    string source_system = 3; // System originating this event_type
+    string source_id = 4; // ID of the originating event in the source system
+    google.protobuf.Timestamp occurred_at = 10;
+    optional string description = 11; // Free-form description of the event
+    optional string actor = 12; // Username triggering the action
+    optional string url = 13; // Link to source event, if available
+
+    string data_type = 14; // MIME type of notification data - usually application/json
+    string data = 15; // Notification data, in format matching specified in  data_type
+
+};
+
+message HappyEvent {
+
+    /*   Happy Event ID
+
+        When creating a new notification, this should not be specified.
+        This field is populated when querying notification records from the
+        database.
+
+    */
+    optional string id = 1;
```

Comment:
Would `HappyEvent` be the struct that we store in the DB or are you thinking it would be wrapped in another struct that gets stored in the DB?

Reply:
HappyEvent wraps Event and describes the record in the database. I'd initially combined them, but I got to thinking we can maybe reuse that core  Event to create something like OktaServiceEvent or whatever. That way it mates up, but we can add fields around it to publish on the bus.

When querying Happy API for event history, you'd get a list of HappyEvents back.

Reply:
I added `created_at` to the `HappyEvent` proto message. Seemed like it could be of interest to a potential calling application.